### ISSUE TITLE
No exit codes upon errors?

### ISSUE THREAD

**Issue by jasonhan-vassar** [vectorgraphics/asymptote#239]
Hi, I've installed Asymptote 2.70 from `brew`. When I use it and an error occurs, no appropriate exit code is raised. For example, `error: unexpected end of input` or `error: could not load module` exits with code 0.

Is this even normal? Is there a way I can know when an error has occurred programmatically? Thanks for reading.

**Comment by johncbowman** [comment]
It works for me. Try
```
echo "{" | asy - || echo error
echo "{}" | asy - || echo error

```

**Comment by jasonhan-vassar** [comment]
Here is what I get:

<img width="745" alt="Screen Shot 2021-04-28 at 1 17 27 PM" src="https://user-images.githubusercontent.com/34594853/116445621-159f9100-a824-11eb-957d-1c9ac74c67ef.png">

This isn't right, is it? `echo error` is not reached in either case.

**Comment by johncbowman** [comment]
It looks like you are maybe running under MacOS? Can you please confirm what OS you are using.
This is an important detail.

**Comment by jasonhan-vassar** [comment]
Yes, I am on macOS 11.1.

**Comment by RubberNoodles** [comment]
@johncbowman I can't seem to reproduce this on macOS 11.0, running on M1 ARM. 
<img width="571" alt="asy-mac" src="https://user-images.githubusercontent.com/69167715/120364596-b1606900-c2ca-11eb-899f-5e7966473dc7.png">

**Comment by johncbowman** [comment]
I can reproduce the problem on a MacBook Pro under Big Sur (Darwin Kernel Version 20.1.0).
It looks like an OS issue; maybe there is an OS patch to fix error code handling?

**Comment by johncbowman** [comment]
Followup: it seems to be a multithreading issue as this correctly generates ERROR:
echo "{" | asy - -nothreads || echo ERROR

**Comment by johncbowman** [comment]
Fixed in commit 76702591d248278c20260206bd1d8f6bff27993a.
